### PR TITLE
Reduce samples dependencies on everything

### PR DESCRIPTION
**Pull request by exFalso**
This is an attempt at reducing the samples' `compile` dependency on `node-driver` and consequently everything else.

This is blocked on #1468 as I couldn't run the demos to completion to test whether everything works.

@Clintonio I also wasn't sure what our artifacts are for demos - is it ok like this?

### CODE REVIEW

File: samples/attachment-demo/build.gradle [hunk 1]
```diff
@@ -26,11 +26,12 @@ dependencies {
     testCompile "junit:junit:$junit_version"
 
     // Corda integration dependencies
-    cordaCompile project(path: ":node:capsule", configuration: 'runtimeArtifacts')
     cordaCompile project(path: ":webserver:webcapsule", configuration: 'runtimeArtifacts')
     cordaCompile project(':core')
-    cordaCompile project(':webserver')
-    cordaCompile project(':node-driver')
+
+    runtime project(path: ":node:capsule", configuration: 'runtimeArtifacts')
```
[the remark this answers is not in the document]
Comment (exFalso) [reply samples/attachment-demo/build.gradle]: This is what I wasn't sure about - we have artifacts for samples, but I'm not sure what they are - shouldn't they just be the cordapps? Do we need the mains published?

[the remark this answers is not in the document]
Comment (andr3ej) [reply samples/attachment-demo/build.gradle]: When running from the IDE, the demos use driver so artifacts aren't needed. But when running from the command line, the demos use NodeRunner so corda.jar and usually also webserver.jar are needed. There is a right way to do this that doesn't involve using a compile dep but I don't know what that right way is, maybe @Clintonio or @chrisr3 does.

[the remark this answers is not in the document]
Comment (exFalso) [reply samples/attachment-demo/build.gradle]: I ran the gradle commands documented on docsite and they still work (to the extent they work on master)

[the remark this answers is not in the document]
Comment (andr3ej) [reply samples/attachment-demo/build.gradle]: The demo commands won't rebuild the jars with your latest changes, that's what motivated the runtime -> compile change in 782.

[the remark this answers is not in the document]
Comment (exFalso) [reply samples/attachment-demo/build.gradle]: You mean they won't be rebuilt on changes to corda? I'll try to add a dependency link without using the compile config

[the remark this answers is not in the document]
Comment (chrisr3) [reply samples/attachment-demo/build.gradle]: I would have thought that

    runtime project('node:capsule')

would have been The Right Thing, from a conceptual point of view at least. Do you need
`configuration: 'runtimeArtifacts'` too. The capsule really shouldn't be on _anyone's_ compile-time classpath.

[the remark this answers is not in the document]
Comment (Clintonio) [reply samples/attachment-demo/build.gradle]: **Artifacts for samples** - Yes, TeamCity builds them. We can change the pattern if they get moved but we want to keep them.

**Correcting the compile dep to runtime** - If it's not a compile dep it doesn't get compiled. There is likely a way to say "I really want this artifact to exist when this task runs" but so far the only obvious way to me was to just have the task _depend_ on the task that creates the artifact. Essentially this whole compile/runtime stuff is from the Java and Kotlin plugins and doesn't really interact with the `deployNodes` task or `Cordformation` plugin at all. Perhaps a task dependency is the right way? I would love to investigate further but time isn't permitting it for me. @exFalso reckon you have time to look into that? All else failing just use a task dependency on `:node:capsule:build:buildCordaJar` or whatever the task is called.

@chrisr3 sort of. This whole runtime dependency thing is a little bit of a hack, as per the above paragraph. Cordformation is just reading from your list of runtime deps and finding the JARs it wants. I've thought of better ways to do it but it won't be done for V1.

**Comment by exFalso** [comment]
Closing this, this was meant for 1.0, but we don't have time for it

**Pull request closed without merging**